cumulusci/tasks/bulkdata/snowfakery.py

```diff
@@ -0,0 +1,657 @@
+      over again until we generate 30,000 new Account records.
+      If the recipe also generates e.g.Contacts, Opportunities or whatever
+      else, it generates the appropriate number of them to match.
+
+      Underscores are allowed but optional in big numbers: 2000000
+      is the same as 2_000_000.
+        """
+        },
+        "run_until_records_loaded": {
+            "description": """<sobject>:<count>
+
+      Run the recipe repeatedly until the number of records of
+      <sobject> uploaded in this task execution matches <count>.
+
+      For example, `--run_until_records_loaded Account:50_000` means:
+
+      Run the recipe over and over again
+      until we generate 50_000 new Account records. If the recipe
+      also generates e.g. Contacts, Opportunities or whatever else, it
+      generates the appropriate number of them to match.
+        """
+        },
+        "run_until_recipe_repeated": {
+            "description": """Run the recipe <count> times,
+            no matter what data is already in the org.
+
+            For example, `--run_until_recipe_repeated 50_000` means
+            run the recipe 50_000 times."""
+        },
+        "working_directory": {
+            "description": "Default path for temporary / working files"
+        },
+        "loading_rules": {
+            "description": "Path to .load.yml file containing rules to use to "
+            "load the file. Defaults to `<recipename>.load.yml`. "
+            "Multiple files can be comma separated."
+        },
+        # "recipe_options": {},  # TODO: Snowfakery 2.1
+        "bulk_mode": {
+            "description": "Set to Serial to force serial mode on all jobs. Parallel is the default."
+        },
+        "drop_missing_schema": {
+            "description": "Set to True to skip any missing objects or fields instead of stopping with an error."
+        },
+        "num_processes": {
+            "description": "Number of data generating processes. Defaults to matching the number of CPUs."
+        },
+    }
+
+    def _validate_options(self):
+        super()._validate_options()
+        # Do not store recipe due to MetaDeploy options freezing
+        recipe = self.options.get("recipe")
+        recipe = Path(recipe)
+        if not recipe.exists():
+            raise exc.TaskOptionsError(f"Cannot find recipe `{recipe}`")
+
+        self.num_generator_workers = self.options.get("num_processes", None)
+
+    # more loader workers than generators because they spend so much time
+    # waiting for responses. 4:1 is an experimentally derived ratio
+    @property
+    def num_loader_workers(self):
+        return self.num_generator_workers * 4
+
+    def setup(self):
+        if not self.num_generator_workers:
+            # logical CPUs do not really improve performance of CPU-bound
+            # code, so we ignore them.
+            self.num_generator_workers = psutil.cpu_count(logical=False)
```

Comment:
Can we use https://docs.python.org/3/library/os.html#os.cpu_count and avoid the new dependency library?

Reply:
Unfortunately it doesn't have the "logical=False" option and therefore it overstates my laptop's CPU count by 100%. Apple claims it is a 4-core machine but Python claims 8.